tests/importable_test.py

```diff
@@ -1,26 +1,100 @@
 """Smoke tests related to loading entry points."""
 
+import typing
 from importlib.metadata import entry_points as _discover_entry_points
 
 import pytest
 
 
 @pytest.mark.parametrize(
-    'entry_points_group',
     (
-        'awx.credential_plugins',
-        'awx_plugins.credentials',
+        'entry_points_group',
+        'expected_entry_point_name',
+        'expected_entry_point_module',
+        'expected_entry_point_func',
     ),
+    [
+        (
+            'awx_plugins.credentials',
+            'aim',
+            'aim',
+            'aim_plugin',
+        ),
+        (
+            'awx_plugins.credentials',
+            'conjur',
+            'conjur',
+            'conjur_plugin',
+        ),
+        (
+            'awx_plugins.credentials',
+            'hashivault_kv',
+            'hashivault',
+            'hashivault_kv_plugin',
+        ),
+        (
+            'awx_plugins.credentials',
+            'hashivault_ssh',
+            'hashivault',
+            'hashivault_ssh_plugin',
+        ),
+        (
+            'awx_plugins.credentials',
+            'azure_kv',
+            'azure_kv',
+            'azure_keyvault_plugin',
+        ),
+        (
+            'awx_plugins.credentials',
+            'centrify_vault_kv',
+            'centrify_vault',
+            'centrify_plugin',
+        ),
+        (
+            'awx_plugins.credentials',
+            'thycotic_dsv',
+            'dsv',
+            'dsv_plugin',
+        ),
+        (
+            'awx_plugins.credentials',
+            'thycotic_tss',
+            'tss',
+            'tss_plugin',
+        ),
+        (
+            'awx_plugins.credentials',
+            'aws_secretsmanager_credential',
+            'aws_secretsmanager',
+            'aws_secretmanager_plugin',
+        ),
+    ],
 )
-def test_entry_points_exposed(entry_points_group: str) -> None:
-    """Verify the plugin entry point is discoverable.
+class TestEntryPoints:
```

Comment:
Could you avoid introducing classes? They only complicate things. The tests are already namespaced by being in modules.

If you want to reuse parametrization, you can assign the marker to a variable and use it.
As in
`with_credential_entry_points = pytest.mark.parametrize(...)`
And then, apply `@with_credential_entry_points` to the tests w/o creating another layer of indirection.